messaging/src/main/java/org/axonframework/eventhandling/MultiSourceTrackingToken.java

```diff
@@ -0,0 +1,194 @@
+
+        MultiSourceTrackingToken otherMultiToken = (MultiSourceTrackingToken) other;
+
+        Assert.isTrue(otherMultiToken.trackingTokens.keySet().equals(this.trackingTokens.keySet()),
+                      () -> "MultiSourceTrackingTokens contain different keys");
+
+        Map<String, TrackingToken> tokenMap = new HashMap<>();
+
+        otherMultiToken.trackingTokens.forEach((key, otherToken) ->
+                                                       tokenMap.put(key, trackingTokens.get(key).upperBound(otherToken))
+        );
+
+        return new MultiSourceTrackingToken(tokenMap);
+    }
+
+    /**
+     * Compares this token to {@code other} checking each member token with its counterpart to see if they are covered
+     * in the {@code other} token
+     *
+     * @param other The token to compare to this one
+     * @return {@code true} if this token covers the other, otherwise {@code false}
+     */
+    @Override
+    public boolean covers(TrackingToken other) {
+        Assert.isTrue(other instanceof MultiSourceTrackingToken, () -> "Incompatible token type provided.");
+
+        MultiSourceTrackingToken otherMultiToken = (MultiSourceTrackingToken) other;
+
+        Assert.isTrue(otherMultiToken.trackingTokens.keySet().equals(this.trackingTokens.keySet()),
+                      () -> "MultiSourceTrackingTokens contain different keys");
+
+        //as soon as one delegated token doesn't cover return false
+        for (Map.Entry<String, TrackingToken> trackingTokenEntry : trackingTokens.entrySet()) {
+            if (!trackingTokenEntry.getValue().covers(otherMultiToken.trackingTokens
+                                                              .get(trackingTokenEntry.getKey()))) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Advances a single token within the tokenMap
+     *
+     * @param streamName        the stream/source which is being advanced
+     * @param newTokenForStream the token representing the new position of the stream
+     * @return the token representing the current processing position of all streams.
+     */
+    public MultiSourceTrackingToken advancedTo(String streamName, TrackingToken newTokenForStream) {
+        trackingTokens.put(streamName, newTokenForStream);
+        return new MultiSourceTrackingToken(trackingTokens);
+    }
+
+    /**
+     * Return the tracking token for an individual stream
+     *
+     * @param streamName the name of the stream for the tracking token
+     * @return the tracking token for the stream
+     */
+    public TrackingToken getTokenForStream(String streamName) {
+        return trackingTokens.get(streamName);
+    }
+
+    /**
+     * @return the map containing the constituent tokens.
+     */
+    public Map<String, TrackingToken> getTrackingTokens() {
+        return trackingTokens;
+    }
+
+    /**
```

Comment:
I think that for consistency with the code base, we should also provide a body for this bit of JavaDoc, even though it's virtually identical to the `@return` line.